ecosystem/platform/server/app/controllers/nft_offers_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+# Copyright (c) Aptos
+# SPDX-License-Identifier: Apache-2.0
+
+class NftOffersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_nft_offer
+  before_action :offer_dependent_logic
+
+  def show; end
+
+  def update
+    nft = Nft.find_by(user: current_user, nft_offer: @nft_offer)
+
+    # TODO: Compute image_url depending on nft_offer.name.
+    image_url = "https://example.com/#{Random.uuid}"
+    nft ||= Nft.create(user: current_user, nft_offer: @nft_offer, image_url:)
+
+    # TODO: Mint the NFT on devnet and save the address.
```

Comment:
One thing here, is that the mint is fully in the FE; 
[Archive.zip](https://github.com/aptos-labs/aptos-core/files/8856608/Archive.zip)

Reply:
That should be fine, we can add it to the nft#show page.